packages/modules/devices/shelly/counter.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+import logging
+from typing import Optional
+from modules.common import req
+from modules.common.component_state import CounterState
+from modules.common.component_type import ComponentDescriptor
+from modules.common.fault_state import ComponentInfo, FaultState
+from modules.common.store import get_counter_value_store
+from modules.common.simcount._simcounter import SimCounter
+from modules.devices.shelly.config import ShellyCounterSetup
+
+log = logging.getLogger(__name__)
+
+
+class ShellyCounter:
+
+    def __init__(self,
+                 device_id: int,
+                 component_config: ShellyCounterSetup,
+                 address: str,
+                 generation: Optional[int]) -> None:
+        self.component_config = component_config
+        self.sim_counter = SimCounter(device_id, self.component_config.id, prefix="bezug")
+        self.store = get_counter_value_store(self.component_config.id)
+        self.fault_state = FaultState(ComponentInfo.from_component_config(self.component_config))
+        self.address = address
+        self.generation = generation
+
+    def update(self) -> None:
+        power = 0
+        if self.generation == 1:
+            status_url = "http://" + self.address + "/status"
+        else:
+            status_url = "http://" + self.address + "/rpc/Shelly.GetStatus"
+        status = req.get_http_session().get(status_url, timeout=3).json()
+        try:
+            if self.generation == 1:  # shelly3EM
+                meters = status['emeters']
+                # shelly3EM has three meters:
+                for meter in meters:
+                    power = power + meter['power']
+                power = power * -1
+
+                voltages = [status['emeters'][0]['voltage'], status['emeters']
+                            [1]['voltage'], status['emeters'][2]['voltage']]
```

Comment:
```suggestion
                voltages = [status['emeters'][i]['voltage'] for i in range(0,3)]
```
Macht das gleiche, ist aber besser lesbar.